whois-api/src/main/java/net/ripe/db/whois/api/rest/WhoisCrossOriginFilter.java

```diff
@@ -69,7 +88,17 @@ public String getQueryString() {
         }
     }
 
-    private static boolean isCrossOrigin(final HttpServletRequest request) {
-        return StringUtils.isNotEmpty(request.getHeader(HttpHeaders.ORIGIN));
+    private static boolean isOriginHeaderPresent(final HttpServletRequest request) {
+        return StringUtils.isNotEmpty(getHttpOrigin(request));
+    }
+
+    private static String getHttpOrigin(HttpServletRequest request) {
+        return request.getHeader(HttpHeaders.ORIGIN);
+    }
+
+    private static boolean isHostsAllowedForCrossOrigin(final HttpServletRequest request, final String[] allowedHostsforCrossOrigin) {
+        if(!isOriginHeaderPresent(request)) return true;
+
+        return Arrays.stream(allowedHostsforCrossOrigin).anyMatch( host -> host.equalsIgnoreCase(getHttpOrigin(request)));
```

Comment:
So we will have to include the protocol in the `whois.allow.cross.origin.hosts` ? You don't strip it out of the Origin header

Reply:
that's a good idea. I will make that change